tools/block-generator/generator/generate.go

```diff
@@ -739,7 +772,13 @@ func (g *generator) initializeLedger() {
 		fmt.Printf("error making genesis: %v\n.", err)
 		os.Exit(1)
 	}
-	l, err := ledger.OpenLedger(logging.Base(), "block-generator", true, ledgercore.InitState{
+	var prefix string
+	if g.genesisID == "" {
+		prefix = "block-generator"
+	} else {
+		prefix = g.genesisID
+	}
+	l, err := ledger.OpenLedger(logging.Base(), prefix, true, ledgercore.InitState{
```

Comment:
just wanted to point out that supplying `GetDefaultLocal` as configuration to OpenLedger may open you up to underlying changes you don't want/expect. For instance, `StorageEngine` is being added in this [PR](https://github.com/algorand/go-algorand/pull/5212/files), which currently specifies sql, but may specify another engine in the future.

Reply:
Do all storage engines support in memory mode?